src/afa/afa.cc

```diff
@@ -34,29 +36,531 @@ const std::string Mata::Afa::TYPE_AFA = "AFA";
+* over these subresults
+* The result is always an upward closed set!
+* @param node a source set of states
+* @return closed set of nodes
+*/
+ClosedSet<State> Afa::post(Node node) const
+{
+    if(node.empty())
+    {
+        return ClosedSet<State>(ClosedSet<State>::upward_closed, 0, transRelation.size()-1, Nodes{Node{}});
+    }
+    ClosedSet<State> result = ClosedSet<State>(ClosedSet<State>::upward_closed, 0, transRelation.size()-1);
+    for(auto transVec : transRelation[*(node.begin())])
+    {
+        result.insert(post(node, transVec.symb).get_antichain());
+    }
+    return result;
+} // post }}}
+
+/** This function allows us to perfom "post" and use the whole alphabet to perform individual transitions 
+* The output will be represented as a ClosedSet to omit the neccessity to explicitly return all the proper nodes
+* We will perform a transition for each node of the antichain of the given set of nodes and then
+* we perform an union over these subresults
+* The result is always an upward closed set!
+* @param nodes a set of sets of nodes used to perform a transition
+* @return closed set of nodes
+*/
+ClosedSet<State> Afa::post(Nodes nodes) const
+{
+    ClosedSet<State> result(ClosedSet<State>::upward_closed, 0, transRelation.size()-1);
+    for(auto node : nodes)
+    {
+        result.insert(post(node).get_antichain());
+    }
+    return result;
+} // post }}}
+
+// The end of the definitions of the "post" functions
+
+//***************************************************
+//
+// PRE
+//
+// Inspection of the automaton in the backward fashion
+//
+//***************************************************
+
+/** This function inspects an inverse transition relation and returns a vector of pointers
+* according to the given source state and symbol
+* Otherwise, it gives us an empty list which means that the result does not exist. We return pointers
+* to be able to directly change the inverse transition relation if there is a neccessity to add a new transition 
+* @brief performs a transition using a single state and symbol
+* @param src a source state
+* @param symb a symbol used to perform an inverse transition
+* @return a vector of pointers to the inverse results
+*/
+InvTrans::InvResultPtrs Afa::perform_inv_trans(State src, Symbol symb) const
+{
+    for(auto element : this->invTransRelation[src])
+    {
+        if(element.symb == symb)
+        {
+            return element.invResultPtrs;
+        }
+    }
+    return InvTrans::InvResultPtrs();
+}  // perform_inv_trans }}}
+
+/** This function inspects an inverse transition relation and returns a vector of pointers
+* according to the given source states and symbol
+* Otherwise, it gives us an empty list which means that the result does not exist. We return pointers
+* to be able to directly change the inverse transition relation if there is a neccessity to add a new transition 
+* @brief performs a transition using a single state and symbol
```

Comment:
Am I right?
```suggestion
* @brief performs an inverse transition using a single state and symbol
```

Reply:
Yes, that is a mistake, thanks for your suggestion!